File: Examples/Samples/sample.py
# Python sample for tree-sitter demo
from dataclasses import dataclass
from typing import Optional, List

@dataclass
class Node:
    value: int
    left: Optional['Node'] = None
    right: Optional['Node'] = None

def insert(root: Optional[Node], value: int) -> Node:
    if root is None:
        return Node(value)
    if value < root.value:
        root.left = insert(root.left, value)
    elif value > root.value:
        root.right = insert(root.right, value)
    return root

def inorder(root: Optional[Node]) -> List[int]:
    if root is None:
        return []
    return inorder(root.left) + [root.value] + inorder(root.right)

def main():
    values = [5, 3, 7, 1, 4, 6, 8]
    root = None
    for v in values:
        root = insert(root, v)
    print("Sorted:", inorder(root))

if __name__ == "__main__":
    main()

File: Examples/Samples/sample.c
/* C sample for tree-sitter demo */
#include <stdio.h>
#include <stdlib.h>

typedef struct Node {
    int value;
    struct Node *left;
    struct Node *right;
} Node;

Node *node_new(int value) {
    Node *n = (Node *)malloc(sizeof(Node));
    n->value = value;
    n->left = NULL;
    n->right = NULL;
    return n;
}

Node *insert(Node *root, int value) {
    if (root == NULL)
        return node_new(value);
    if (value < root->value)
        root->left = insert(root->left, value);
    else if (value > root->value)
        root->right = insert(root->right, value);
    return root;
}

void inorder(Node *root) {
    if (root == NULL) return;
    inorder(root->left);
    printf("%d ", root->value);
    inorder(root->right);
}

int main(void) {
    int values[] = {5, 3, 7, 1, 4, 6, 8};
    int n = sizeof(values) / sizeof(values[0]);
    Node *root = NULL;
    for (int i = 0; i < n; i++)
        root = insert(root, values[i]);
    inorder(root);
    printf("\n");
    return 0;
}
